tests/networkpolicy_test.go

```diff
@@ -0,0 +1,90 @@
+)
+
+const (
+	privilegedPodYaml  = "tests/manifests/privileged-pod.yml"
+        denyallYaml  = "tests/manifests/networkpolicy-deny-all.yml"
+	allowhttpYaml = "tests/manifests/networkpolicy-allow-http.yml"
+)
+
+
+var _ = Describe("[rfe_id:150][crit:high][vendor:cnv-qe@redhat.com][level:component]Networkpolicy", func() {
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	ktests.PanicOnError(err)
+
+	var vmia *v1.VirtualMachineInstance
+	var vmib *v1.VirtualMachineInstance
+
+	ktests.BeforeAll(func() {
+		ktests.SkipIfUseFlannel(virtClient)
+		ktests.SkipIfNotUseNetworkPolicy(virtClient)
+		ktests.BeforeTestCleanup()
+	})
+
+	Context("Networkpolicy allow http port", func() {
+		It("[test_id:CNV-369] Set connectivity to only allow from http port", func() {
+			By("Create VMIs")
+			userData := "#cloud-config\npassword: fedora\nchpasswd: { expire: False }\n"
+			vmia = ktests.NewRandomVMIWithEphemeralDiskAndUserdata(ktests.ContainerDiskFor(ktests.ContainerDiskFedora), userData)
+			ktests.AddExplicitPodNetworkInterface(vmia)
+			vmib = ktests.NewRandomVMIWithEphemeralDiskAndUserdata(ktests.ContainerDiskFor(ktests.ContainerDiskFedora), userData)
+			ktests.AddExplicitPodNetworkInterface(vmib)
+			vmia.Spec.Domain.Resources.Requests[k8sv1.ResourceName("memory")] = resource.MustParse("1024M")
+			_, err = virtClient.VirtualMachineInstance(ktests.NamespaceTestDefault).Create(vmia)
+			Expect(err).ToNot(HaveOccurred())
+			vmib.Spec.Domain.Resources.Requests[k8sv1.ResourceName("memory")] = resource.MustParse("1024M")
+			_, err = virtClient.VirtualMachineInstance(ktests.NamespaceTestDefault).Create(vmib)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Install httpd and start the service inside the VMI")
+			ktests.WaitUntilVMIReady(vmia, ktests.LoggedInFedoraExpecter)
+			err = ktests.CheckForTextExpecter(vmia, []expect.Batcher{
+				&expect.BSnd{S: "\n"},
+				&expect.BExp{R: "#"},
+				&expect.BSnd{S: "sudo yum install httpd -y\n"},
+				&expect.BExp{R: "#"},
+				&expect.BSnd{S: "echo test | sudo tee /var/www/html/index.html\n"},
+				&expect.BExp{R: "test"},
+				&expect.BSnd{S: "sudo systemctl start httpd\n"},
+				&expect.BExp{R: "#"},
+				&expect.BSnd{S: "curl localhost:80\n"},
+				&expect.BExp{R: "test"},
+			}, 180)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Create the networkpolicy")
+			tests.CreateResourceWithFilePath(denyallYaml)
+			tests.CreateResourceWithFilePath(allowhttpYaml)
+
+			By("Validate the http port is available")
+			vmia, err = virtClient.VirtualMachineInstance(ktests.NamespaceTestDefault).Get(vmia.Name, &v13.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			ip := vmia.Status.Interfaces[0].IP
+			ktests.WaitUntilVMIReady(vmib, ktests.LoggedInFedoraExpecter)
+			err = ktests.CheckForTextExpecter(vmib, []expect.Batcher{
+				&expect.BSnd{S: fmt.Sprintf("curl %s:80 \n", ip)},
+				&expect.BExp{R: "test"},
+			}, 60)
+			Expect(err).ToNot(HaveOccurred())
+		})
```

Comment:
The test only covers the "allow-tcp" policy, and its result would be the same even if you didn't
 apply the "deny-by-default" policy.
I think you should add a negative step here - try to connect to a port other than the allowed port (80), and expect it to fail.

Reply:
Thanks for review. Yes, I was planning to do that, but I think the deny-all policy already covered by https://github.com/kubevirt/kubevirt/blob/master/tests/networkpolicy_test.go#L85

Reply:
In the test you mentioned, the "deny-by-default" is covered as a standalone. The new test you composed should verify the  "deny-by-default" can co-exist with the "allow-tcp" policy. This is what is done in the [Polarion test](https://polarion.engineering.redhat.com/polarion/#/project/CNV/workitem?id=CNV-369) which is the source for this automation.
In your new test, you do apply the "deny-by-default" policy, so I think it is preferred to verify it is applied as expected, by trying to connect to a port !=80, and expect this connaction attempt to fail.

Reply:
I agree with Yossi here, I think we should make sure that only port 80 is opened and we cannot access different port.